SalesforceCommerceCloud/pwa-kit
# @W-15654904Showing strikethrough price on PLP

# Description
This PR implemented strikethrough price for products on PLP if any product has a listing sale price 
![image](https://github.com/SalesforceCommerceCloud/pwa-kit/assets/52219283/d0831e9a-4dfc-4dcc-adfc-5ec77cd300af)

**NOTE**: You may notice the price may/ may not display correctly in Recommendation section/Recently View. It will be fix later in a different ticket, please ignore. 




# Types of Changes




> Breaking changes include:
>
> - Removing a public function or component or prop
> - Adding a required argument to a function
> - Changing the data type of a function parameter or return value
> - Adding a new peer dependency to `package.json`

# Changes

- (change1)

# How to Test-Drive This PR

- Checkout the code
- npm ci at root
- start retail app
- Confirm price display for different type of products
```
* Standard product/Bundle: since there are no variations always show the exact list price, and a strikethrough price (if promo or price-book adjustment) and callout 
    * No swatches
    * No range of price
    * Show whatever is the LOWER sale price


* Variant products: 
    * The initially loaded product tile image should be the representedProduct
    * We will display swatches for color variants of products with that variant type
        * Customers should be able to configure (constants or config? or just override?) the variant facet used to populate the swatches because color is just one possible variant type employed by users
    * On initial load:
        * Display a price range “From X Y” inclusive of all the variants of that product
            * X is the lowest variant price (including price book sale adjustments & promotional discounts)
            * Y is that given lowest priced variant’s list price
        * When a filter is applied or a variant is selected
            * If we have a single variant show the exact price
            * If we have multiple variants show a range
        * This should be abstracted in such a way that logic should be overridable
    * We will show the callout message for the promotionId that provides the greatest possible, non-conditional discount for the shown variant
    * Product Tile should link the users to the selected variant from PLP to PDP
* Product sets: 
    * The initially loaded product tile will be the master product
    * We will NOT EVER display a swatch
    * Show From X where X is the lowest price among the set children
    * We will show the callout message for the promotionId that provides the greatest possible, non-conditional discount

```
- Master product on first load
![image](https://github.com/SalesforceCommerceCloud/pwa-kit/assets/52219283/075aacb4-cb63-4cf4-9290-7a6a31a32348)

- Bundle
![image](https://github.com/SalesforceCommerceCloud/pwa-kit/assets/52219283/fabd5944-bbdb-45e1-9bce-bc308aa7d667)
 

- Standard 
![image](https://github.com/SalesforceCommerceCloud/pwa-kit/assets/52219283/7e2f2cd2-a9a8-47e2-9689-88ec46a21ee2)

- Product set
![image](https://github.com/SalesforceCommerceCloud/pwa-kit/assets/52219283/f8d9eca4-ae6d-440c-8d43-d2738447ca29)

# Checklists




## General


## Accessibility Compliance

You must check off all items in **one** of the follow two lists:


_or..._


## Localization
